tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/ActionResponseFilter.cs

```diff
@@ -0,0 +1,32 @@
+// <copyright file="ActionResponseFilter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+using Datadog.Trace.DuckTyping;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class ActionResponseFilter : IActionFilter
+{
+    public void OnActionExecuting(ActionExecutingContext context)
+    {
+    }
+
+    public void OnActionExecuted(ActionExecutedContext context)
+    {
+        var security = Security.Instance;
+        if (security.Enabled && context.Result.TryDuckCast<ObjectResult>(out var result))
```

Comment:
I wonder, does this duck typing make sense here 🤔 Given you're directly referencing Microsoft.AspNetCore.Mvc can't you just do this?

```suggestion
        if (security.Enabled && context.Result is Microsoft.AspNetCore.Mvc.ObjectResult result)
```

Reply:
yes, so `ObjectResult` is in `Microsoft.AspNetCore.Mvc.Core.dll` and we don't reference it.. I didn't want to add more Microsoft.AspNetCore.* references to what we already have  🤔